src/analyze/flavors/lwc/utils.ts

```diff
@@ -0,0 +1,142 @@
+
+const LWCCACHE = Symbol("LWC Compoeent");
+export function isLwcComponent(node: Node, context: AnalyzerVisitContext): ComponentRef | undefined {
+	if ((node as any)[LWCCACHE]) {
+		return (node as any)[LWCCACHE];
+	}
+	const r = _isLwcComponent(node, context);
+	if (r) {
+		(node as any)[LWCCACHE] = r;
+	}
+	return r;
+}
+
+function _isLwcComponent(node: Node, context: AnalyzerVisitContext): ComponentRef | undefined {
+	const { ts } = context;
+	if (ts.isClassDeclaration(node)) {
+		const jsName = node.getSourceFile().fileName;
+
+		const splitjsName = jsName.split("/");
+		if (splitjsName.length < 3) {
+			return;
+		}
+		const nameSpace = splitjsName[splitjsName.length - 3];
+		const componentName = splitjsName[splitjsName.length - 2];
+		const tagName = nameSpace + "-" + camelToDashCase(componentName);
+
+		// Main case (~100% of the cases)
+		// The class is a default export and there is a template (.html) starting with <template>
+		// Moreover the JS file name should match the directory name, minus the extension (js|ts)
+		//    https://lwc.dev/guide/reference#html-file
+		const flags = getCombinedModifierFlags(node);
+		if (flags & ModifierFlags.ExportDefault && (jsName.endsWith(".js") || jsName.endsWith(".ts"))) {
+			const fileName = splitjsName[splitjsName.length - 1];
+			const fileNoExt = fileName.substring(0, fileName.length - 3);
+			if (fileNoExt === componentName) {
+				const htmlName = jsName.substring(0, jsName.length - 3) + ".html";
+				if (existsSync(htmlName)) {
+					const content = readFileSync(htmlName, "utf8").trim();
+					if (content.startsWith("<template>")) {
+						return { tagName };
+					}
+				}
+			}
+		}
+
+		// Edge case
+		// The components are not matching the file naming recommendations, so we check the inheritance
+		const lightning = inheritFromLightning(node, context);
+		if (lightning) {
+			return { tagName };
+		}
+
+		// Finally, we use a JS doc definition in case none of the above work
+		// This is the last resort
+		const v = parseJsDocForNode(
+			node,
+			["lwc-element"],
+			(tagNode, { name }) => {
+				return { tagName: name || tagName };
+			},
+			context
+		);
+		if (v && v.length === 1) {
+			return v[0] as ComponentRef;
+		}
+	}
+
+	return undefined;
+}
+
+// Check if the Class inherits from lighning
+// For now, we just check one level
+function inheritFromLightning(node: ClassDeclaration, context: AnalyzerVisitContext): boolean {
+	const { checker } = context;
+	if (node.heritageClauses) {
+		for (const clause of node.heritageClauses) {
+			// OK we are getting strange results here with the token beeing 87 (ElseKeyword), 89 (ExportKeyword)
+			// Not sure why for now, so we skip checking the keyword as 'LightningElement' is dicriminant enough
+			//if (clause.token == SyntaxKind.ExtendsKeyword) {
+			const symbol = checker.getSymbolAtLocation(clause.types[0].expression);
+			if (symbol?.escapedName === "LightningElement") {
+				return true;
+			}
+			//}
+		}
+	}
```

Comment:
Note that I'm working on a solution so that it becomes possible to easily use existing functionality to analyze the entire chain of inheritance (with support for resolving mixins eg). This functionality would also be very valuable when analyzing `LitElement` and `FASTElement`.

Reply:
This is a great addition. LWC is using its own module resolution (https://rfcs.lwc.dev/rfcs/lwc/0020-module-resolution) so we'll need to make sure that this is pluggable.